bokehjs/src/coffee/models/glyphs/hex_tile.ts

```diff
@@ -0,0 +1,279 @@
+      ysize *= this.model.aspect_scale
+    }
+    else {
+      xsize /= this.model.aspect_scale
+    }
+
+    const points = []
+    for (let i = 0; i < this._x.length; i++) {
+      const x = this._x[i]
+      const y = this._y[i]
+      if (isNaN(x+y) || !isFinite(x+y))
+        continue
+      points.push({minX: x-xsize, minY: y-ysize, maxX: x+xsize, maxY: y+ysize, i})
+    }
+    return new RBush(points)
+  }
+
+  // overriding map_data instead of _map_data because the default automatic mappings
+  // for other glyphs (with cartesian coordinates) is not useful
+  map_data(): void {
+    const self = this as any
+
+    [self.sx, self.sy] = this.map_to_screen(self._x, self._y);
+
+    [self.svx, self.svy] = this._get_unscaled_vertices()
+
+  }
+
+  protected _get_unscaled_vertices(): [number[], number[]] {
+    const size = this.model.size
+    const aspect_scale = this.model.aspect_scale
+
+    if (this.model.orientation == "pointytop") {
+      const rscale = this.renderer.yscale
+      const hscale = this.renderer.xscale
+
+      const r = Math.abs(rscale.compute(0) - rscale.compute(size))                               // assumes linear scale
+      const h = Math.sqrt(3)/2*Math.abs(hscale.compute(0) - hscale.compute(size)) / aspect_scale // assumes linear scale
+      const r2 = r/2.0
+
+      const svx = [0, -h,  -h,   0,  h,  h ]
+      const svy = [r,  r2, -r2, -r, -r2, r2]
+
+      return [svx, svy]
+    }
+
+    else {
+      const rscale = this.renderer.xscale
+      const hscale = this.renderer.yscale
+
+      const r = Math.abs(rscale.compute(0) - rscale.compute(size))                               // assumes linear scale
+      const h = Math.sqrt(3)/2*Math.abs(hscale.compute(0) - hscale.compute(size)) * aspect_scale // assumes linear scale
+      const r2 = r/2.0
+
+      const svx = [r,  r2, -r2, -r, -r2, r2]
+      const svy = [0, -h,  -h,   0,  h,  h ]
+
+      return [svx, svy]
+    }
+
+  }
+
+  protected _render(ctx: Context2d, indices: number[], {sx, sy, scale}: HexTileData): void {
+
+    const [svx, svy] = this._get_unscaled_vertices()
+
+    for (const i of indices) {
+      if (isNaN(sx[i] + sy[i] + scale[i]))
+        continue;
+
+      ctx.translate(sx[i], sy[i])
+      ctx.beginPath();
+      for (let j = 0; j < 6; j++) {
+        ctx.lineTo(svx[j]*scale[i], svy[j]*scale[i])
+      }
+      ctx.closePath()
+      ctx.translate(-sx[i], -sy[i])
+
+      if (this.visuals.fill.doit) {
+        this.visuals.fill.set_vectorize(ctx, i);
+        ctx.fill();
+      }
+
+      if (this.visuals.line.doit) {
+        this.visuals.line.set_vectorize(ctx, i);
+        ctx.stroke();
+      }
+
+    }
+  }
+
+  protected _hit_point(geometry: PointGeometry): Selection {
+    const {sx, sy} = geometry
+    const x = this.renderer.xscale.invert(sx)
+    const y = this.renderer.yscale.invert(sy)
+
+    const candidates = this.index.indices({minX: x, minY: y, maxX: x, maxY: y})
+
+    const hits = Array()
```

Comment:
Idiomatic code should read `const hits = []`. This isn't incorrect, but it catches the eye (maybe the author missed an initializer, etc.).